packages/veritone-react-common/src/components/ContentTemplates/ContentTemplateForm/index.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { string, shape, objectOf, any, func } from 'prop-types';
+import Button from '@material-ui/core/Button';
+import withMuiThemeProvider from 'helpers/withMuiThemeProvider';
+import ContentTemplates from 'components/ContentTemplates';
+
+import styles from './styles.scss';
+
+@withMuiThemeProvider
+export default class ContentTemplateForm extends React.Component {
+  static propTypes = {
+    templateData: objectOf(
+      shape({
+        id: string,
+        name: string.isRequired,
+        status: string,
+        definition: objectOf(any)
+      })
+    ).isRequired,
+    initialTemplates: objectOf(
+      shape({
+        id: string,
+        name: string.isRequired,
+        status: string,
+        definition: objectOf(any),
+        data: objectOf(any)
+      })
+    ),
+    onSubmit: func.isRequired
+  };
+
+  static defaultProps = {
+    initialTemplates: {}
+  };
+
+  state = {
+    contentTemplates: {}
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  UNSAFE_componentWillMount() {
+    const newState = {
+      contentTemplates: { ...this.props.initialTemplates }
+    };
+
+    this.setState(newState);
+  }
+
+  manageTemplatesList = (templateSchemaId, remove = false) => {
+    const { templateData, initialTemplates } = this.props;
+
+    if (remove) {
+      if (this.state.contentTemplates[templateSchemaId]) {
+        return this.setState(prevState => {
+          const contentTemplates = { ...prevState.contentTemplates };
+          delete contentTemplates[templateSchemaId];
+
+          return { contentTemplates };
+        });
+      }
+    } else {
+      const data = {};
+      Object.keys(templateData[templateSchemaId].definition.properties).reduce(
+        (fields, schemaDefProp) => {
+          data[schemaDefProp] =
```

Comment:
fyi you can simplify this to `data[schemaDefProp] = get(initialTemplates, `${templateSchemaId}.data.${schemaDefProp}`, '')`